add interface to specify YARN application name

kmaehashi opened the pull request:
This patch improves the Jubatus-on-YARN API to accept externally-specified YARN application name.
This makes operators easy to distinguish each JoY job in management tools like YARN Resource Manager Web UI.

rimms commented:
I faced compile error:

```
[info] Set current project to jubatus-on-yarn-client (in build file:/path/to/jubatus-on-yarn/jubatusonyarn/)
[info] Compiling 7 Scala sources to /path/to/jubatus-on-yarn/jubatusonyarn/jubatus-on-yarn-client/target/scala-2.10/classes...
[error] /path/to/jubatus-on-yarn/jubatusonyarn/jubatus-on-yarn-client/src/main/scala/us/jubat/yarn/client/Model.scala:31: overloaded method value start with alternatives:
[error]   (aLearningMachineName: String,aLearningMachineType: us.jubat.yarn.common.LearningMachineType,aZookeepers: List[us.jubat.yarn.common.Location],aConfigFile: org.apache.hadoop.fs.Path,aResource: us.jubat.yarn.client.Resource,aNodeCount: Int,aBasePath: org.apache.hadoop.fs.Path,aApplicationName: String)scala.concurrent.Future[us.jubat.yarn.client.JubatusYarnApplication] <and>
[error]   (aLearningMachineName: String,aLearningMachineType: us.jubat.yarn.common.LearningMachineType,aZookeepers: List[us.jubat.yarn.common.Location],aConfigFile: org.apache.hadoop.fs.Path,aResource: us.jubat.yarn.client.Resource,aNodeCount: Int,aApplicationName: String)scala.concurrent.Future[us.jubat.yarn.client.JubatusYarnApplication] <and>
[error]   (aLearningMachineName: String,aLearningMachineType: us.jubat.yarn.common.LearningMachineType,aZookeepers: List[us.jubat.yarn.common.Location],aConfigString: String,aResource: us.jubat.yarn.client.Resource,aNodeCount: Int,aBasePath: org.apache.hadoop.fs.Path,aApplicationName: String)scala.concurrent.Future[us.jubat.yarn.client.JubatusYarnApplication] <and>
[error]   (aLearningMachineName: String,aLearningMachineType: us.jubat.yarn.common.LearningMachineType,aZookeepers: List[us.jubat.yarn.common.Location],aConfigString: String,aResource: us.jubat.yarn.client.Resource,aNodeCount: Int,aApplicationName: String)scala.concurrent.Future[us.jubat.yarn.client.JubatusYarnApplication]
[error]  cannot be applied to (String, us.jubat.yarn.common.LearningMachineType.Classifier.type, List[us.jubat.yarn.common.Location], org.apache.hadoop.fs.Path, us.jubat.yarn.client.Resource, Int)
[error]   val tApplicationFuture = JubatusYarnApplication.start(
[error]                                                   ^
[error] /path/to/jubatus-on-yarn/jubatusonyarn/jubatus-on-yarn-client/src/main/scala/us/jubat/yarn/client/YarnClientServlet.scala:59: overloaded method value startJubatusApplication with alternatives:
[error]   (aName: String,aLearningMachineType: us.jubat.yarn.common.LearningMachineType,aZookeepers: List[us.jubat.yarn.common.Location],aConfigFile: org.apache.hadoop.fs.Path,aResource: us.jubat.yarn.client.Resource,aNodeCount: Int,aBasePath: org.apache.hadoop.fs.Path,aApplicationName: String)us.jubat.yarn.client.ApplicationMasterProxy <and>
[error]   (aName: String,aLearningMachineType: us.jubat.yarn.common.LearningMachineType,aZookeepers: List[us.jubat.yarn.common.Location],aConfigString: String,aResource: us.jubat.yarn.client.Resource,aNodeCount: Int,aBasePath: org.apache.hadoop.fs.Path,aApplicationName: String)us.jubat.yarn.client.ApplicationMasterProxy
[error]  cannot be applied to (String, us.jubat.yarn.common.LearningMachineType, List[us.jubat.yarn.common.Location], String, us.jubat.yarn.client.Resource, Int, org.apache.hadoop.fs.Path)
[error]       mController.get.startJubatusApplication(tSettings.learningMachineInstanceName, tSettings.learningMachineType, tListZooKeeper, tSettings.configJson, tSettings.resources, tSettings.nodeCount, new Path(tSettings.basePath))
[error]                       ^
[error] two errors found
[error] (jubatus-on-yarn-client/compile:compile) Compilation failed
```

kmaehashi commented:
I revised the contents of this PR.
- YARN application name for JoY can now be specified externally.
- The old (four overloaded) `start` APIs had too many arguments.  I added new `start` API that accepts cluster configurations via one object, and deprecated old ones.
- Previously, list of ZooKeepers are given as `List[Location]`, however this was inappropriate, as ZooKeeper location string is a kind of DSL parsed by ZooKeeper client library.  I fixed this to directly pass `String` instead.
- Cluster launching options are now well-validated.
- Removed `/start` HTTP API in `YarnClientServlet`, which was unused.
